src/lib/registration/widgets/registration_code.rb

```diff
@@ -67,6 +64,16 @@ def register
           return true
         end
 
+        if Registration.is_registered?
+          log.info("The system is already registered so skipped registration.")
+          return true
+        end
+
+        if !SwMgmt.find_base_product
```

Comment:
This check is just a safe check, we could rely that always that the widget is used there exists a base_product but just be safer.